### PR TITLE
[WFLY-19932] Use Testcontainers to start Kafka rather than running em…

### PR DESCRIPTION
**Pull request by kabir**
…bedded"

https://issues.redhat.com/browse/WFLY-19932

### CODE REVIEW

File: testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/reactive/KeystoreUtil.java [hunk 1]
```diff
@@ -25,10 +26,16 @@ public class KeystoreUtil {
...
     public static final String SERVER_KEYSTORE = KEY_STORE_DIRECTORY + "/server.keystore.p12";
     public static final Path SERVER_KEYSTORE_PATH = Paths.get(SERVER_KEYSTORE);
+
+    public static final String SERVER_KEYSTORE_CREDENTIALS = KEY_STORE_DIRECTORY + "/server.keystore.pwd.txt";
+
+    public static final Path SERVER_KEYSTORE_CREDENTIALS_PATH = Paths.get(SERVER_KEYSTORE_CREDENTIALS);
+
     private static final String SERVER_CER = KEY_STORE_DIRECTORY + "/server.cer";
-    private static final String SERVER_TRUSTSTORE = KEY_STORE_DIRECTORY + "/server.truststore.p12";
+    private static final String SERVER_TRUSTSTORE = KEY_STORE_DIRECTORY + "/git ";
```
Comment (rhusar) [inline testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/reactive/KeystoreUtil.java:38]: Something is off here. The change seems inadvertent - and also, there is a trailing space.

**Review by rhusar — CHANGES_REQUESTED** [review]

**Review by rhusar — CHANGES_REQUESTED** [review]
With this change, since it's using containers, the test will have to be excluded on Windows for now.

    java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration

File: testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/reactive/KeystoreUtil.java (hunk 1 above)
Comment (kabir) [reply testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/reactive/KeystoreUtil.java:38]: Thanks, I must have been trying to run terminal commands while in intellij :-)

**Review by rhusar — CHANGES_REQUESTED** [review]

File: testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/reactive/messaging/kafka/compression/ReactiveMessagingKafkaWithSnappyFailsOnWindowsAndMacTestCase.java [hunk 2]
```diff
@@ -31,6 +32,7 @@
 @RunWith(Arquillian.class)
 @RunAsClient
 @ServerSetup({RunKafkaSetupTask.class, EnableReactiveExtensionsSetupTask.class})
+@DockerRequired
```
[the remark this answers is not in the document]
Comment (rhusar) [reply testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/reactive/messaging/kafka/compression/ReactiveMessagingKafkaWithSnappyFailsOnWindowsAndMacTestCase.java:35]: So @kabir this is now merged, so no need to change this. Ideally just rebase.

**Comment by kabir** [comment]
Thanks @rhusar  rebased

**Pull request merged**